From 33b4a6199535868dba8446e5191223e83ea3fe0f Mon Sep 17 00:00:00 2001
From: Matt Turner <mattst88@gmail.com>
Date: Mon, 8 Apr 2024 12:14:00 -0400
Subject: [PATCH] =?UTF-8?q?test:=20Raise=20SIGALRM=20interval=20to=2050?=
 =?UTF-8?q?=C2=B5s?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On some platforms, an interval of 5µs is short enough that the test
spends its time almost exclusively processing SIGALRMs and never
progresses otherwise. Raising the interval to 50µs allows the test to
pass in a fraction of a second.

Bug: https://bugs.gentoo.org/916777
Closes: https://gitlab.freedesktop.org/libinput/libei/-/issues/50
---
 test/eierpecken.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/eierpecken.c b/test/eierpecken.c
index 2f2e225d..7eb60ce1 100644
--- a/test/eierpecken.c
+++ b/test/eierpecken.c
@@ -521,11 +521,11 @@ new_context(enum peck_ei_mode ei_mode)
 			struct itimerval timer = {
 				.it_interval = {
 					.tv_sec = 0,
-					.tv_usec = 5,
+					.tv_usec = 50,
 				},
 				.it_value = {
 					.tv_sec = 0,
-					.tv_usec = 5,
+					.tv_usec = 50,
 				}
 			};
 			rc = xerrno(setitimer(ITIMER_REAL, &timer, 0));
-- 
GitLab

